src/Orchard.Environment.Extensions/ExtensionLibraryService.cs

```diff
@@ -45,16 +43,21 @@ public class ExtensionLibraryService : IExtensionLibraryService
             IOptions<ExtensionProbingOptions> optionsAccessor,
             ILogger<ExtensionLibraryService> logger)
         {
-            _metadataReferences = new Lazy<List<MetadataReference>>(GetMetadataReferences);
             _hostingEnvironment = hostingEnvironment;
             _probingDirectoryName = optionsAccessor.Value.DependencyProbingDirectoryName;
             _probingFolderPath = _hostingEnvironment.ContentRootFileProvider.GetFileInfo(Path.Combine(optionsAccessor.Value.RootProbingName, _probingDirectoryName)).PhysicalPath;
             _logger = logger;
         }
 
-        public IEnumerable<MetadataReference> MetadataReferences()
-        {
-            return _metadataReferences.Value;
+        public IEnumerable<string> MetadataPaths {
+            get {
+                return _compileOnlyAssemblies
+                    .Values
+                    .Concat(
+                        _loadedAssemblies
+                            .Values
+                            .Select(x => x.Value.Location));
+            }
```

Comment:
I've seen you now pass metadata references to the razor view engine as `MetadataReferenceFeature` through `ApplicationPartManager.FeatureProviders`. Wow, did you try it, does it work? If so, cool.

So, what i can say is that normally here we exclude assemblies which are ambient (part of the app). This by using an hashset which is initialized with `ApplicationAssemblyNames` (see below), not to add them but to exclude them.